### DISCUSSION TITLE
Joi Resolver does not stop submit

### DISCUSSION THREAD

**Discussion by Deccoyi** [react-hook-form/react-hook-form#9723] [Q&A] [answered]
Hi I am usin JoiResolver to validate my form. I am using rsuite UI library. Its input components doesnt really work alone. I had to use `Controller` to get the inputs. I have two datepickers startDate and endDate. Here is the scheme I created

```javascript
const EventSchema = Joi.object({
    title: Joi.string().alphanum().required(),
    startDate: Joi.date(),
    endDate: Joi.date().greater(Joi.ref('startDate'))
    // endDate: Joi.date().greater(Joi.ref('start'))
})
```

When I leave the title field empty and press submit. Everything works. Error message gets displayed and form doesnt gets submitted. But when I select an earlier endDate and submit this happens: 
![image](https://user-images.githubusercontent.com/17047360/210792603-ace72a45-e0b9-41c1-96cd-c59ceffecd27.png)

What am I missing? Is it a bug or do I need a workaround?

If I use isValid from` formState : {isValid}` and disable the Submit button it works. But form doesn't re-render. So error message is not shown.

**Comment by Deccoyi** [comment]
For those who has similar problem I quit using Joi. instead, I used `rules` object to write my own validation functions. But that didn't work as well. So problem is not Joi resolver. Problem is probably between rsuite-rhf . But I have no ida why. 

My Solution (not a perfect one. Maybe a workaround): Just like I said in my question `isValid` is working properly. But it doesnt re render the form and error messages doesn't Show up. I changed my submit button to something like this

```javascript
{isValid ? (<Button type="submit"> Save </Button>) : (<Button onClick={() => {trigger()} }> Save </Button>)}
```

So if there is an error on the form it doesnt get submitted. And `trigger()` makes sure it is rerendered and error messages show up.

**Comment by leapful** [comment]
@Deccoyi 

Please provide a codesandbox url to re-produce the issue.

**Reply by Deccoyi** [reply]
In my project it doesn't give " Invalid argument supplied to oneOf, expected an array." warning but on codesanbox it does. I'm not sure why. But the error e.focus is not a function is the error I have.

https://codesandbox.io/s/rsuite-form-vss6n0?file=/src/App.js

**Comment by leapful** [comment, marked as answer]
@Deccoyi 

Your problem is not related to Joi resolver but from `Input` and `DatePicker` component of UI library `rsuite` which support prop `ref` as DOM reference handler therefore the DOM element of each component cannot be registered correctly to be used in case focus on error validation.

For case `Input` component, it can be achieved with its custom prop `inputRef`.

For case `DatePicker` component, it's a little bit more hidden settings to register DOM reference as this component is an organism component that combines multiple components and behaviors. The prop `ref` of DatePicker is now registered for an instance of React component, not DOM reference therefore you need to use field `target` of this instance as DOM reference to focus.

Working demo: https://codesandbox.io/s/rsuite-form-forked-2r762p?file=/src/App.js

**Reply by Deccoyi** [reply]
Thank you. I quit using JOI but kept the title same. After I quit, I realized that it is because of rsuite but it was too late to delete the topic  :)
